src/main/java/io/supertokens/multitenancy/Multitenancy.java

```diff
@@ -375,23 +375,25 @@ public static boolean addUserIdToTenant(Main main, TenantIdentifierWithStorage t
                 .addUserIdToTenant(tenantIdentifierWithStorage, userId);
     }
 
-    public static boolean removeUserIdFromTenant(Main main, TenantIdentifierWithStorage tenantIdentifierWithStorage, String userId)
+    public static boolean removeUserIdFromTenant(Main main, TenantIdentifierWithStorage tenantIdentifierWithStorage, String userId, String externalUserId)
             throws FeatureNotEnabledException, TenantOrAppNotFoundException, StorageQueryException,
             UnknownUserIdException {
         if (Arrays.stream(FeatureFlag.getInstance(main, new AppIdentifier(null, null)).getEnabledFeatures())
                 .noneMatch(ee_features -> ee_features == EE_FEATURES.MULTI_TENANCY)) {
             throw new FeatureNotEnabledException(EE_FEATURES.MULTI_TENANCY);
         }
 
-        boolean finalDidExist = false;
+
         boolean didExist = AuthRecipe.deleteNonAuthRecipeUser(tenantIdentifierWithStorage, userId);
-        finalDidExist = finalDidExist || didExist;
+
+        if (externalUserId != null) {
+            didExist = AuthRecipe.deleteNonAuthRecipeUser(tenantIdentifierWithStorage, externalUserId) || didExist;
+        }
```

Comment:
if external user id is not null, you can just call deleteNonAuthRecipeUser once with the external user ID, or else call it with the user ID. No need to call this twice.

This is cause, all non auth recipe stuff is against the external user id only.